gaia-media-controls.js

```diff
@@ -0,0 +1,291 @@
+;(function(define){'use strict';define(function(require,exports,module){
+
+/**
+ * Dependencies
+ */
+var Component = require('gaia-component');
+var MediaControlsImpl = require('./lib/media-controls-impl');
+
+var MediaControls = Component.register('gaia-media-controls', {
+  /**
+   * 'createdCallback' is called when the element is first created.
+   */
+  created: function() {
+    console.log('creating gaia-media-controls web component...');
+  },
+
+  attachTo: function(player) {
+
+    if (this.mediaPlayerImpl) {
+      throw new Error('A media player is already attached to the media controls component');
+    }
+
+    if (!this.shadowRoot) {
+      this.setupShadowRoot();
+    }
+    this._impl = new MediaControlsImpl(this, this.shadowRoot, player);
+  },
+
+  detachFrom: function() {
+    if (this.mediaPlayerImpl) {
+      this.mediaPlayerImpl.unload();
+      this.mediaPlayerImpl = null;
+    }
+  },
+
+  /*
+   * Expose testing helper functions
+   */
+  enableComponentTesting() {
+    if (this._impl) {
+      this._impl.enableComponentTesting();
+
+      var componentTestingHelper = {
+        triggerEvent:
+          this._impl.triggerEvent.bind(this._impl),
+        getElement:
+          this._impl.getElement.bind(this._impl),
+        disableComponentTesting:
+          this._impl.disableComponentTesting.bind(this._impl)
+      };
+    }
+
+    return componentTestingHelper;
+  },
+
+  template: `
+
+  <style>
+
+  @font-face {
+  	font-family: "gaia-icons";
+  	src: url("fonts/gaia-icons.ttf") format("truetype");
```

Comment:
How is this going to work? Is the .ttf file going to be part of this component's repo?

I see that gaia-header has an explicit require('gaia-icons') call in it.  I suspect you should do that, or just document that this component assumes that the gaia-icons font is already loaded.

In that case, you can leave out the @font-face declaration and just use the font, right?